### PR TITLE
Feat profile show

### PR DESCRIPTION
**Pull request by Jonrw97**
# Profile show & edit

This adds the profile show page, profile edit form(using turbo frames) and creates a profile for a new user when signing up via GitHub

### Profile show

Screenshot 2023-09-14 141336

### Profile edit

Screenshot 2023-09-14 141359

## Mobile
Screenshot 2023-09-13 182515
Screenshot 2023-09-13 182534
Screenshot 2023-09-13 182551

### CODE REVIEW

File: app/models/profile.rb [hunk 1]
```diff
@@ -7,8 +7,6 @@ class Profile < ApplicationRecord
   has_many :requests, dependent: :destroy
   has_many :favourites, dependent: :destroy
```
Comment (dewaldreynecke) [inline app/models/profile.rb:9]: This probably needs:

```suggestion
has_many :profile_languages, dependent: :destroy
```

File: app/views/profiles/_preview.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<turbo-frame id=<%= profile.id %>>
+  <div class="card">
+    <div class="card-header">
+    </div>
+    <div class="card-content">
+      <div class="two-column">
+        <div>
+          <%= image_tag profile.user.avatar_url, class:"profile-avatar" %>
```
Comment (dewaldreynecke) [inline app/views/profiles/_preview.html.erb:8]: We might have to wrap this avatar for safety – what if there isn't one present? The Github API call may have failed, or in the future we might want to give the user the option to remove their avatar. 

A default placeholder type image may be a good idea?

**Review by dewaldreynecke — CHANGES_REQUESTED** [review]
Looks good. I like the use of Turbo frames, my new favourite thing.
Just two possible improvements I think – check my other comments.

File: app/models/profile.rb (hunk 1 above)
Comment (Jonrw97) [reply app/models/profile.rb:9]: This should be on line 3 already

Comment (dewaldreynecke) [reply app/models/profile.rb:9]: My apologies – those lines were collapsed in my view here on Github. All good.

**Comment by Jonrw97** [comment]
@dewaldreynecke I have done a basic check to see if there is a GitHub avatar URL, not much we can do if the URL is broken from GitHub(which would hopefully be a rare occurrence). There is something you can do with jQuery that will display an alternative image instead of text but I need to investigate more. This should work for the time being.

**Review by dewaldreynecke — APPROVED** [review]

**Comment by dewaldreynecke** [comment]
Seems good thanks. Merging this now.

**Pull request merged**